qutebrowser/browser/urlmarks.py

```diff
@@ -244,17 +264,35 @@ class BookmarkManager(UrlMarkManager):
         - removed gets emitted with the URL as argument.
     """
 
+    def _bookmarks_directory(self):
+        """Return the directory containing the bookmarks.
+
+        For bookmarks are data, return by default a folder inside the standard
+        data directory. In order not to break the compatibility with an older
+        behavior, if bookmarks are found in the standard configuration
+        directory, return this directory instead.
+        """
+        config_dir = standarddir.config()
+        if config_dir is not None:
+            inconfigdir = os.path.join(standarddir.config(), 'bookmarks')
+        else:
+            inconfigdir = None
+        indatadir = os.path.join(standarddir.data(), 'bookmarks')
+        if inconfigdir is not None and os.path.isdir(inconfigdir):
+            return inconfigdir
+        else:
+            return indatadir
```

Comment:
Keeping the return statements at the end of the method is useless and unreadable -- there should be only one branch deciding which path to use, not two. What about this?

```python
        config_dir = standarddir.config()
        if config_dir is not None:
            inconfigdir = os.path.join(config_dir, 'bookmarks')
            if os.path.isdir(inconfigdir):
                return inconfigdir
        return os.path.join(standarddir.data(), 'bookmarks')
```

On second thought, if `standarddir.config()` can return `None`, are we sure that `standarddir.data()` doesn't?